src/renderer/components/BalanceGraph/BalanceGraph.tsx

```diff
@@ -56,92 +56,64 @@ export const BalanceGraph: React.FunctionComponent<IBalanceGraphProps> = (props:
         interval===IntervalValue ? selector="selected" : null;
         return `graph-option ${selector}`;
     };
-    // let refreshInterval: number = 0;
+
+    const awaitData = async (): Promise<void>=>{
+        props.getData(interval).then((dataValueArray)=>{
+            setXAxis(dataValueArray, interval);
+        });
+    };
+
     useEffect(()=>{
-    
         clearInterval(refreshInterval);
-        console.log("ON MOUNT: " + refreshInterval);
 
         const timeNow = new Date();
-        console.log(timeNow);
+        const msUntil = (60-(timeNow.getSeconds()))*1000;
+        const minUntil = ((60-(timeNow.getMinutes()-1))*1000*60)+msUntil;
+        const hourUntil = ((24-(timeNow.getHours()-1))*1000*60*60)+minUntil;
+
         switch (interval) {
             case IntervalEnum.HOUR: {
-                const seconds = timeNow.getSeconds();
-                console.log("seconds now: " + seconds);
-            
                 const intervalHandler = (): void =>{
                     awaitData();
                     const localRefreshInterval = window.setInterval(awaitData, 60000);
                     setRefreshInterval(localRefreshInterval);
                 };
-
-                window.setTimeout(intervalHandler,(60-seconds)*1000);
-
-                console.log("Case HOUR, refreshInterval on case: " + refreshInterval);
+                window.setTimeout(intervalHandler,msUntil);
```

Comment:
there is method called setInterval: https://www.w3schools.com/jsref/met_win_setinterval.asp

It will return reference, keep this reference and on unMounting of component clearInterval